erpnext/accounts/utils.py

```
@@ -2724,3 +2730,353 @@ def build_qb_match_conditions(doctype, user=None) -> list:
+	if cfg.get("check_batch_availability"):
+		_check_batch_availability_warn(doc)
+
+	if cfg.get("check_stock_availability"):
+		_check_stock_availability_warn(doc)
+
+
+def _check_prev_docstatus(doc):
+	try:
+		if hasattr(doc, "check_prev_docstatus"):
+			doc.check_prev_docstatus()
+	except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)